projects/filjs-audit/README.md

```diff
@@ -0,0 +1,651 @@
+
+#### Experiments
+
+Explorations:
+
+- [connect-lotus-client-with-deployed-lotus-instance.md](https://github.com/protocol/w3dt-sudo/blob/master/home/vasco-santos/2021-02-exploration-report-connect-lotus-client-with-deployed-lotus-instance.md)
+- [js-lotus-client.md](https://github.com/protocol/w3dt-sudo/blob/master/home/vasco-santos/2021-03-exploration-report-1-on-js-lotus-client.md)
+
+### [Filecoin.js](https://github.com/filecoin-shipyard/filecoin.js)
+
+> A JavaScript (and TypeScript) library for interacting with the Filecoin's Lotus node, with support for external signers.
+
+This work was funded by a Devgrant, intending to be familiar to [ethers.js](https://github.com/ethers-io/ethers.js/) users, but development was paused due to lack of adoption and some difficulties with the message handling/signing workflow.
+
+#### Maturity
+
+The current state of the library specified in the README is `ALPHA: things will not work or work incorectly, will break and the API will change! ⚠️`
+
+Taking into account the activity of the repo, it has not been in actively development since November 2020, relying on an outdated fork of `filecoin-signing-tools-js`. There is a [Roadmap Issue](https://github.com/filecoin-shipyard/filecoin.js/issues/1) where we can see that general Filecoin operations are no yet implemented.
+
+The latest version in npm is `"0.0.5-alpha"`, however the `package.json` in GitHub shows `"0.0.3-alpha"`, so there may be more recent code not pushed to GitHub.
+
+Last updated: 17 Nov 2020
+
+#### Key Findings
+
+- Supports Wallet Mnemonics
+- The roadmap indicates that more general Filecoin operations might get implemented in the future.
+- Tests are failing to run, even by trying to run a shell script as done in CI. In the latest commits, CI did also not run.
+
+#### Recommendations
+
+* ***(NEEDS INVESTIGATION)*** Look into what is happening with the versioning and whether there's code not pushed to GitHub.
+
+_TODO: are these recommendations?_
+
+- Exported `WsJsonRpcConnector` and `HttpJsonRpcConnector` seem the same as [@filecoin-shipyard/lotus-client-provider-nodejs](@filecoin-shipyard/lotus-client-provider-nodejs) + [@filecoin-shipyard/lotus-client-provider-browser](@filecoin-shipyard/lotus-client-provider-browser), which abstracts the WS or HTTP, but also supports both.
+  - It is worth pointing out that even in the browser, `node-fetch` seems to be bundled and every time the window existence is checked.
+- Interact with traditional wallet providers like metamask (could not get it up and running)
+  - Needs a frictionless experience or a general improvement on the documentation for this core functionality provided
+
+#### Experiments
```

Comment:
This section seems like the weak point of the report at the moment because it's so much detail compared to the rest. Maybe we could trim this down a bit and link back to the original exploration report? I don't think we want to remove all of it because there's interesting stuff in here.

Reply:
(actually the real "weak point" is all the unfilled sections, the things we didn't even look at! but in terms of the structure and content of the meaty bits of the report this stands out as different due to its density).

Reply:
yeah, I will link this to the individual exploration report.

> actually the real "weak point" is all the unfilled sections

Yeah, I could not find a better structure for those

Reply:
👌 I certainly don't want to diminish the work and the content is great but the risk is that we overload this with content and nobody reads any of it at all